### ISSUE TITLE
haxelib run + cwd behaviour

### ISSUE THREAD

**Issue by nadako** [HaxeFoundation/haxelib#272]
The current behaviour is quite weird: We append cwd to the end of argument list and then change cwd to the library installation directory. What's the reason it's implemented like that? cc @ncannasse 

I think it would be more correct not to change cwd and set the lib directory to the `HAXELIB_RUN` env var (currently it's always set to `1`).

Changing this will break current haxelibs, so if we want to do it, we should think of some way to stay compatible.

**Comment by nadako** [comment]
Assigning to @ncannasse for comments :)

**Comment by ncannasse** [comment]
I agree with the change. Not sure how to make sure to handle backward compatibility.

**Comment by nadako** [comment]
One possibility is to add some "haxelib protocol version" field to haxelib.json and use the new mechanism if it's set to proper value (and warn if author submits a library using old mechanism). But I'm not sure it's worth adding that field just for this change.
